packages/pigeon/example/app/ios/Runner/Messages.g.swift

```diff
@@ -222,6 +223,30 @@ class ExampleHostApiSetup {
     } else {
       sendMessageChannel.setMessageHandler(nil)
     }
+    let sendMessageModernAsyncChannel = FlutterBasicMessageChannel(
+      name:
+        "dev.flutter.pigeon.pigeon_example_package.ExampleHostApi.sendMessageModernAsync\(channelSuffix)",
+      binaryMessenger: binaryMessenger, codec: codec)
+    if let api = api {
+      sendMessageModernAsyncChannel.setMessageHandler { message, reply in
+        let args = message as! [Any?]
+        let messageArg = args[0] as! MessageData
+        Task {
+          do {
+            let result = try await api.sendMessageModernAsync(message: messageArg)
+            DispatchQueue.main.async {
```

Comment:
you can do `Task { @MainActor }` here.

Reply:
> you can do `Task { @MainActor }` here.

Do you mean:

```swift
            sendMessageModernAsyncChannel.setMessageHandler { message, reply in
                let args = message as! [Any?]
                let messageArg = args[0] as! MessageData
                Task {
                    do { @MainActor
                        let result = try await api.sendMessageModernAsync(message: messageArg)
                        reply(wrapResult(result))

                    } catch {
                        reply(wrapError(error))
                    }
                }
            }
```

Will it lead to executing `api.sendMessageModernAsync` on main thread that could block it?

Reply:
No, i meant you can replace `DispatchQueue.main.async {}` with `Task { @MainActor}`

Reply:
> No, i meant you can replace `DispatchQueue.main.async {}` with `Task { @MainActor}`

What's the point/profit?

Reply:
The purpose of swift concurrency is to replace GCD. So since you are using swift concurrency here, there's no reason to use GCD at all.

Reply:
> GCD

Okay I get it.

I am not very familiar with swift concurrency actually. I've tried to do as you asked and this is not valid syntax:

![image](https://github.com/user-attachments/assets/af897043-800d-457d-bedc-e8fc1f539291)
 
Maybe I could use `MainActor.run` instead?

```swift
sendMessageModernAsyncChannel.setMessageHandler { message, reply in
    let args = message as! [Any?]
    let messageArg = args[0] as! MessageData
    Task {
        do {
            let result = try await api.sendMessageModernAsync(message: messageArg)

            MainActor.run(){
                reply(wrapResult(result))
            }

        } catch {
            MainActor.run(){
                reply(wrapError(error))
            }
        }
    }
}
```

Reply:
In your first chunk of code, don't put `await` in front of `Task`. 

For this particular case, you can use `MainActor.run { ... }`, since `reply` doesn't require an async context.

Reply:
@hellohuanlin done